controllers/snapshot/snapshot_adapter.go

```diff
@@ -80,6 +82,11 @@ func (a *Adapter) EnsureAllIntegrationTestPipelinesExist() (results.OperationRes
 			"IntegrationTestScenarios", len(*integrationTestScenarios))
 		for _, integrationTestScenario := range *integrationTestScenarios {
 			integrationTestScenario := integrationTestScenario //G601
+			if integrationTestScenario.Spec.Environment.Name != "" && len(integrationTestScenario.Spec.Environment.Configuration.Env) != 0 {
```

Comment:
I think we also need to check if the `integrationTestScenario.Spec.Environment` is defined before checking the `Name` because the `Environment` field isn't required.

Reply:
done

Reply:
Yep, looks good.